sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -115,11 +116,11 @@ class StatefulProcessorHandleImpl(store: StateStore, runId: UUID)
 
   def getHandleState: StatefulProcessorHandleState = currState
 
-  override def getValueState[T](stateName: String): ValueState[T] = {
+  override def getValueState[K, T](stateName: String, keyEncoder: Encoder[K]): ValueState[T] = {
```

Comment:
Definitely would be better to provide this for users, because the type of the key is known this time and we even have a key encoder in KeyValueGroupedDataset. 

AFAIK, FlatMapGroupsWithState passes StateEncoder over driver to executor. Maybe you need to extract out serializer and deserializer expressions to serialize over the wire but at least it is technically feasible you can pass the keyEncoder rather than requiring users to provide every time.

We can file a JIRA ticket to deal with this, as this PR is holding for a while and we have more works to do before calling to be done. I won't block this PR due to this.

Reply:
Sure - created JIRA here - https://issues.apache.org/jira/browse/SPARK-46852